builder.go

```diff
@@ -134,7 +134,7 @@ func check(resources []Resource) error {
 				return err
 			}
 			if _, ok := cache[dep.FromResource]; !ok {
-				return fmt.Errorf("Dependent resource %s doesn't exist. Validation failure.", dep.FromResource)
+				return fmt.Errorf("Dependent resource %s doesn't exist", dep.FromResource)
```

Comment:
why not create strong typed errors then? you can take some message as input to it, but make it `graph.DependencyFailedError`. Also, you can add a method called `ShouldRetry()` to make this explicit.

Reply:
the idea was to encompass all Validation errors as one error type and the caller can do

```
if err != nil {
  if _, ok := err.(graph.ValidationError) {
    // don't retry
  else {
    // retry
  }
}
```

i believe having a ShouldRetry() method would require all errors returned from the library to become typed and to have that method which seems like more effort than it's worth, no?

Reply:
ok.. Still think this can be improved a bit.. not a blocker